core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -242,18 +242,53 @@ public TransportClientNodesService removeTransportAddress(TransportAddress trans
         if (closed) {
             throw new IllegalStateException("transport client is closed");
         }
-        ensureNodesAreAvailable(nodes);
+
+
+        doRetryWithExponentialBackoff(
+              () -> {
+                  ensureNodesAreAvailable(nodes);
+              },
+              (e) -> {
+                  throw e;
+              });
+
         int index = getNodeNumber();
         RetryListener<Response> retryListener = new RetryListener<>(callback, listener, nodes, index, hostFailureListener);
         DiscoveryNode node = retryListener.getNode(0);
-        try {
-            callback.doWithNode(node, retryListener);
-        } catch (Exception e) {
+
+        doRetryWithExponentialBackoff(
+              () -> {
+                  callback.doWithNode(node, retryListener);
+              },
+              (e) -> {
+                  try {
+                      //this exception can't come from the TransportService as it doesn't throw exception at all
+                      listener.onFailure(e);
+                  } finally {
+                      retryListener.maybeNodeFailed(node, e);
+                  }
+              });
+    }
+
+    public void doRetryWithExponentialBackoff(BasicCallback mainAttempt, ExceptionHandlingCallback onFailure) {
+        int failure = 0;
+        int maxFailure = 6;
+        while (true) {
             try {
-                //this exception can't come from the TransportService as it doesn't throw exception at all
-                listener.onFailure(e);
-            } finally {
-                retryListener.maybeNodeFailed(node, e);
+                mainAttempt.execute();
+                return;
+            } catch (RuntimeException mainException) {
+                try {
+                    failure += 1;
+                    if (failure <= maxFailure) {
+                        Thread.sleep(((int) Math.pow(2, failure)) * 1000);
```

Comment:
It might be worth logging the failure just so we can see what the startup delay is being caused by

Reply:
Agree, I can only do printStackTrace here though